app/models/concerns/order_filtering.rb

```diff
@@ -0,0 +1,96 @@
+# Filtering and priority logic for orders is extracted here to avoid cluttering the model class
+module OrderFiltering
+  extend ActiveSupport::Concern
+
+  def is_priority?
+    self.class.priority.exists?(self)
+  end
+
+  module ClassMethods
+    def priority
+      join_order_transports.where <<-SQL
+        (state = 'submitted' AND submitted_at::timestamptz <= timestamptz '#{one_day_ago}') OR
+        (state = 'processing' AND processed_at::timestamptz < timestamptz '#{last_6pm}') OR
+        (state = 'awaiting_dispatch' AND order_transports.scheduled_at::timestamptz < timestamptz '#{Time.now}') OR
+        (state = 'dispatching' AND dispatch_started_at::timestamptz < timestamptz '#{last_6pm}')
+      SQL
+    end
+
+    #
+    # Returns orders filtered using the following options :
+    #   - states[] A list of string matching the 'state' column of the order
+    #   - types[] A list of custom types, each defined by a combination or properties
+    #   - priority? Whether to only return orders deemed as 'high priority' or not
+    #
+    # Available types to filter on
+    #   - appointment
+    #   - online_order_pickup
+    #   - online_order_ggv
+    #   - shipment
+    #   - other
+    #
+    # To add a custom type, add an sql returning method formatted as {type_name}_sql
+    # e.g :
+    #   def shipment_sql
+    #     "detail_type = 'shipment'"
+    #   end
+    #
+    def filter(states: [], types: [], priority: false)
+      res = all
+      res = res.where("state IN (?)", states) unless states.empty?
+      res = res.where_types(types) unless types.empty?
+      res = res.priority if priority.present?
```

Comment:
This can be written as res.priority.presence. This way be can avoid 1 extra check.